devops/scripts/benchmarks/html/scripts.js

```diff
@@ -0,0 +1,982 @@
+    layerComparisonsData = processLayerComparisonsData(benchmarkRuns);
+    allRunNames = [...new Set(benchmarkRuns.map(run => run.name))];
+
+    // Set up active runs
+    const runsParam = getQueryParam('runs');
+    if (runsParam) {
+        const runsFromUrl = runsParam.split(',');
+
+        // Start with an empty set
+        activeRuns = new Set();
+
+        // Process each run from URL
+        runsFromUrl.forEach(run => {
+            if (run === 'default') {
+                // Special case: include all default runs
+                (defaultCompareNames || []).forEach(defaultRun => {
+                    if (allRunNames.includes(defaultRun)) {
+                        activeRuns.add(defaultRun);
+                    }
+                });
+            } else if (allRunNames.includes(run)) {
+                // Add the specific run if it exists
+                activeRuns.add(run);
+            }
+        });
+    } else {
+        // No runs parameter, use defaults
+        activeRuns = new Set(defaultCompareNames || []);
+    }
+
+    // Setup UI components
+    setupRunSelector();
+    setupSuiteFilters();
+    setupTagFilters();
+    setupToggles();
+
+    // Apply URL parameters
+    const regexParam = getQueryParam('regex');
+    const suitesParam = getQueryParam('suites');
+    const tagsParam = getQueryParam('tags');
+
+    if (regexParam) {
+        document.getElementById('bench-filter').value = regexParam;
+    }
+
+    if (suitesParam) {
+        const suites = suitesParam.split(',');
+        document.querySelectorAll('.suite-checkbox').forEach(checkbox => {
+            checkbox.checked = suites.includes(checkbox.getAttribute('data-suite'));
+        });
+    }
+
+    // Apply tag filters from URL
+    if (tagsParam) {
+        const tags = tagsParam.split(',');
+        tags.forEach(tag => {
+            const checkbox = document.querySelector(`.tag-checkbox[data-tag="${tag}"]`);
+            if (checkbox) {
+                checkbox.checked = true;
+                activeTags.add(tag);
+            }
+        });
+    }
+
+    // Setup event listeners
+    document.querySelectorAll('.suite-checkbox').forEach(checkbox => {
+        checkbox.addEventListener('change', filterCharts);
+    });
+    document.getElementById('bench-filter').addEventListener('input', filterCharts);
+
+    // Draw initial charts
+    updateCharts();
+}
+
+// Make functions available globally for onclick handlers
+window.addSelectedRun = addSelectedRun;
+window.removeRun = removeRun;
+window.toggleAllTags = toggleAllTags;
+
+// Load data based on configuration
+function loadData() {
+    const loadingIndicator = document.getElementById('loading-indicator');
+    loadingIndicator.style.display = 'block'; // Show loading indicator
+
+    if (typeof remoteDataUrl !== 'undefined' && remoteDataUrl !== '') {
+        // Fetch data from remote URL
+        fetch(remoteDataUrl)
+            .then(response => response.json())
```

Comment:
Should check the status of the response object here: afaik `fetch()` will still construct a response if the response was e.g. status 404 -- an error needs to be manually thrown in order to be caught by `.catch()`.
```suggestion
            .then(response => {
                if (!response.ok) { throw new Error(`Got response status ${response.status}.`) }
                return response.json();
            })
```

Reply:
... I think I'd be the one fixing this though, so I've gone ahead and just done it 😅